Leetcode/EigenDemo/eigenPractice.cpp

```diff
@@ -0,0 +1,103 @@
+    std::cout << "print matrix 2x3: " << std::endl;
+    for (int i = 0; i < 2; i++) {
+        for (int j = 0; j < 3; j++) std::cout << matrix_23(i, j) << "\t";
+        std::cout << std::endl;
+    }
+
+    v_3d << 3, 2, 1;
+    vd_3d << 4, 5, 6;
+
+    Matrix<double, 2, 1> result = matrix_23.cast<double>() * v_3d;
+    std::cout << "matrix_23 * v_3d = " << result.transpose() << std::endl;
+
+    matrix_33 << 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9;
+    std::cout << "random matrix: \n" << matrix_33 << std::endl;
+    std::cout << "transpose: \n" << matrix_33.transpose() << std::endl;
+    std::cout << "sum: " << matrix_33.sum() << std::endl;
+    std::cout << "trace: " << matrix_33.trace() << std::endl;
+    std::cout << "times 10: \n" << 10 * matrix_33 << std::endl;
+    std::cout << "inverse: \n" << matrix_33.inverse() << std::endl;
+    std::cout << "det: " << matrix_33.determinant() << std::endl;
+
+    SelfAdjointEigenSolver<Matrix3d> eigen_solver(matrix_33.transpose() * matrix_33);
+    std::cout << "Eigen values = \n" << eigen_solver.eigenvalues() << std::endl;
+    std::cout << "Eigen vectors = \n" << eigen_solver.eigenvectors() << std::endl;
+
+    //matrix_NN * x = v_Nd
+    Matrix<double, MATRIX_SIZE, MATRIX_SIZE> matrix_NN
+    = MatrixXd::Random(MATRIX_SIZE, MATRIX_SIZE);
+    matrix_NN = matrix_NN * matrix_NN.transpose();
+    Matrix<double, MATRIX_SIZE, 1> v_Nd = MatrixXd::Random(MATRIX_SIZE, 1);
+
+    clock_t time_stt = clock();
+    Matrix<double, MATRIX_SIZE, 1> x = matrix_NN.inverse() * v_Nd;
+    std::cout << "time of normal inverse is " << 1000 * (clock() - time_stt) / (double) CLOCKS_PER_SEC << "ms" << std::endl;
+    std::cout << "x = " << x.transpose() <<std::endl;
+
+    time_stt = clock();
+    x = matrix_NN.colPivHouseholderQr().solve(v_Nd);
+    std::cout << "time of Qr decomposition is " << 1000 * (clock() - time_stt) / (double) CLOCKS_PER_SEC << "ms" << std::endl;
+    std::cout <<"x = " << x.transpose() << std::endl;
+
+    time_stt = clock();
+    x = matrix_NN.ldlt().solve(v_Nd);
+    std::cout << "time of ldlt decomposition is " << 1000 * (clock() - time_stt) / (double) CLOCKS_PER_SEC << "ms" << std::endl;
+    std::cout <<"x = " << x.transpose() << std::endl;
+}
+
+void eigenPractice::eigenGeometry() {
+    Matrix3d rotation_matrix = Matrix3d::Identity();
+    AngleAxisd rotation_vector(M_PI / 4, Vector3d(0, 0, 1));
+    std::cout.precision(3);
+    std::cout << "rotation matrix = \n" << rotation_vector.matrix() << std::endl;
+    
+    rotation_matrix = rotation_vector.toRotationMatrix();
+    Vector3d v(1, 0, 0);
+    Vector3d v_rotated = rotation_vector * v;
+    std::cout << "(1, 0, 0) after rotation (by angle axis) = " << v_rotated.transpose() << std::endl;
+    v_rotated = rotation_matrix * v;
+    std::cout << "(1, 0, 0) after rotation (by matrix) = " << v_rotated.transpose() << std::endl;
+    
+    Vector3d euler_angles = rotation_matrix.eulerAngles(2, 1, 0); // ZYX
```

Comment:
```suggestion
    Vector3d euler_angles = rotation_matrix.eulerAngles(2, 1, 0);  // ZYX
```

https://google.github.io/styleguide/cppguide.html#Implementation_Comments

These end-of-line comments should be separated from the code by 2 spaces.